src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +665,36 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private IReadOnlyList<PackageSource> GetAllPackageSources()
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => { return await GetAllPackageSourcesAsync(); });
```

Comment:
Are these causing UI delays right now?

I wouldn't be surprised if they were, but they are just not high enough hit count for us to notice them.

Reply:
I haven't checked, it wouldn't surprise me as a possibility as we go to disk for this information but from my experience these sources are usually pretty small.